### DISCUSSION TITLE
Fail to build PyCall when installing PySR on a Linux server

### DISCUSSION THREAD

**Discussion by Nithouson** [MilesCranmer/PySR#466] [Q&A]
I fail to install PySR on my Linux server because PyCall cannot be built. I got the following error:
`LoadError: Couldn't find libpython; check your PYTHON environment variable.`

I have started an [issue](https://github.com/JuliaPy/PyCall.jl/issues/1061) on the PyCall.jl repository, yet no one answered for over 10 days. Have anyone encouter similar problems when installing PySR, or any idea how to fix this? (See my issue for PyCall for the environment info and full output message. )

Thanks!

**Comment by MilesCranmer** [comment]
Hm, I wonder if this is the issue?

```
gnu/libpython3.10.a"\n/usr/local/lib/python3.10/config-3.10-x86_64-linux-gnu/libpython3.10.a: invalid ELF header")
dlopen(/usr/local/lib/libpython3.10.a) ==> ErrorException("could not load library "/usr/local/lib/libpython3.10.a"\n/usr/local/lib/libpython3.10.a: invalid ELF header")
```

**Reply by Nithouson** [reply]
I think probably you're right. So any idea how to fix this?
I didn't modify the file `libpython3.10.a` after installing Python 3.10. It should be as is.

**Reply by MilesCranmer** [reply]
It almost sounds like that file is corrupted. Otherwise why is the ELF header invalid...

**Reply by MilesCranmer** [reply]
Not sure if any of these are useful: https://pyjulia.readthedocs.io/en/latest/troubleshooting.html

**Reply by MilesCranmer** [reply]
You could also just try running `dlopen` on that file from Julia. Once you have a libpython that works for `dlopen`, should be good

**Reply by stevengj** [reply]
No, the `.a` file is not a shared library, so that will fail.   It needs to find a shared library (`.so` file)

**Reply by MilesCranmer** [reply]
Right, sorry, I'm mixing up shared/static libraries vs binaries.

@Nithouson can you find a `libpython*.so*` file anywhere?

**Reply by Nithouson** [reply]
I can find `libpython3.6.so`, but not such a file for Python 3.10 (which I am using). 
```
find -name libpython*.so
./usr/lib/python3.6/config-3.6m-x86_64-linux-gnu/libpython3.6m.so
./usr/lib/python3.6/config-3.6m-x86_64-linux-gnu/libpython3.6.so
./usr/lib/x86_64-linux-gnu/libpython3.6m.so
```
```
find -name libpython*.a
./usr/lib/python3.6/config-3.6m-x86_64-linux-gnu/libpython3.6m.a
./usr/lib/python3.6/config-3.6m-x86_64-linux-gnu/libpython3.6m-pic.a
./usr/lib/x86_64-linux-gnu/libpython3.6m.a
./usr/local/lib/python3.10/config-3.10-x86_64-linux-gnu/libpython3.10.a
./usr/local/lib/libpython3.10.a
```

**Reply by MilesCranmer** [reply]
How did you install python? I like pyenv
